operator/controllers/manifest_controller.go

```diff
@@ -428,3 +426,24 @@ func (r *ManifestReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Mana
 		}).
 		Complete(r)
 }
+
+func (r *ManifestReconciler) finalizeDeletion(ctx context.Context, manifestObj *v1alpha1.Manifest) error {
+	// remove finalizer
+	finalizerRemoved := controllerutil.RemoveFinalizer(manifestObj, labels.ManifestFinalizer)
+
+	// finally update Manifest, if finalizer was removed
+	if finalizerRemoved {
+		// delete remote cluster information if present
+		if manifestObj.Spec.Remote {
+			kymaOwnerLabel, err := internalUtil.GetKymaLabel(manifestObj)
+			if err != nil {
+				return err
+			}
+			r.clusterCache.Delete(client.ObjectKey{Name: kymaOwnerLabel, Namespace: manifestObj.Namespace})
```

Comment:
The cluster cache does not need to be removed once the Manifest is gone. Rather it would have to be removed once All Manifests for a given Cluster a gone. Currently the client will be kept in memory for eternity in lifecycle-manager, and we could only optimizing it by revoking the key on kyma deletion. Similarly here we would have to only delete it if all manifests are gone.

Reply:
yes I had this is in mind. For deletion of cache entry it is not very straight forward and we could take 2 approaches (I implemented the 2nd):

- during deletion of a Manifest check, filter by label if other Manifests with same Kyma name label exist
- the current implementation, when one Manifest CR is being deleted the Kyma entry is simply deleted and created again, when another Manifest with the deleted Kyma name entry is being reconciled

We need to decide, what have a lower overhead.

Reply:
I will implement the first option above in another commit and then we can decide on the strategy.

Reply:
The main problem i see with the first strategy is mass deletion. If lots of manifests are deleted this will lead to memory spikes as the clients get created again and again during reconcile loops. Lets try label based and then see if it works out fine. In the meantime, i also added the deletion eviction on the module catalog pr in lifecycle manager to cover the same case.